initial-ai-src/script.js

```diff
@@ -0,0 +1,587 @@
+class RuterDiscountCalculator {
+  constructor() {
+    this.maxDiscount = 40; // Maximum discount percentage
+
+    // Zone-based single ticket prices (NOK)
+    this.zonePrices = {
+      1: 39, // Zone 1 only
+      2: 55, // Zones 1-2
+      3: 73, // Zones 1-3
+      4: 91, // Zones 1-4
+      all: 109, // All zones
+    };
+
+    // Zone-based period ticket prices (NOK)
+    this.periodTickets = {
+      1: {
+        day: { price: 110, name: "24-hour ticket", days: 1 },
+        week: { price: 280, name: "7-day ticket", days: 7 },
+        month: { price: 790, name: "30-day ticket", days: 30 },
+        year: { price: 7900, name: "Annual ticket", days: 365 },
+      },
+      2: {
+        day: { price: 155, name: "24-hour ticket", days: 1 },
+        week: { price: 395, name: "7-day ticket", days: 7 },
+        month: { price: 1110, name: "30-day ticket", days: 30 },
+        year: { price: 11100, name: "Annual ticket", days: 365 },
+      },
```

Comment:
🔧  A type definition would help keep this consistent. You could've used either TypeScript or jsdoc comments to ensure the types are right. For example: misspelling days, price, or month would've led to runtime error. Runtime errors like that don't need to exist in 2025. Use types in future.